command-line-interface.md

```diff
@@ -0,0 +1,208 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
+That executable MUST support commands with the following template:
+
+```sh
+$ funC [global-options] <COMMAND> [command-specific-options] <command-specific-arguments>
+```
+
+## Global options
+
+None are required, but the runtime MAY support options that start with at least one hyphen.
+Global options MAY take positional arguments (e.g. `--log-level debug`).
+Command names MUST NOT start with hyphens.
+The option parsing MUST be such that `funC <COMMAND>` is unambiguously an invocation of `<COMMAND>` (even for commands not specified in this document).
+If the runtime is invoked with an unrecognized command, it MUST exit with a nonzero exit code and MAY log a warning to stderr.
+
+## Character encodings
+
+This API specification does not cover character encodings, but runtimes SHOULD conform to their native operating system.
+For example, POSIX systems define [`LANG` and related environment variables][posix-lang] for [declaring][posix-locale-encoding] [locale-specific character encodings][posix-encoding], so a runtime in an `en_US.UTF-8` locale SHOULD write its [version](#version) to stdout in [UTF-8][].
+
+## Commands
+
+### version
+
+Print the runtime version and exit.
+
+* *Options* None are required, but the runtime MAY support options.
+* *Standard streams*
+    * *stdin:* The runtime MUST NOT attempt to read from its stdin.
+    * *stdout:* The runtime MUST print its name, a space, and its version as the first line to its stdout.
+      The name MAY contain any Unicode characters, but MUST NOT contain control codes or newlines.
+      The runtime MAY print additional lines to its stdout, and the format for those lines is not specified in this document.
+    * *stderr:* The runtime MAY print diagnostic messages to stderr, and the format for those lines is not specified in this document.
+* *Exit code:* The runtime MUST exit with zero.
+
+#### Example
+
+```
+$ funC version
+funC 1.0.0
+Built for x86_64-pc-linux-gnu
+$ echo $?
+0
+```
+
+### create
+
+[Create][create] a container from a [bundle directory][bundle].
+This command SHOULD exit as quickly as possible after creating the container, without waiting for the container process to exit.
```

Comment:
On Wed, Jul 20, 2016 at 12:13:29PM -0700, Mrunal Patel wrote:
> > +### create
> > +
> > +[Create][create] a container from a [bundle directory][bundle].
> > +This command SHOULD exit as quickly as possible after creating the container, without waiting for the container process to exit.
> 
> can we remove `as quickly as possible` from the wording?

I'm not wild about that either, but couldn't come up with a better
approach (more details in the commit message for 08af1f6c).  Just
dropping it leaves you with noise, because *all* successful ‘create’
calls are going to exit at some point after creating the container.